Sources/Sentry/SentryFileManager.m

```diff
@@ -746,6 +746,15 @@ - (void)createPathsWithOptions:(SentryOptions *)options
         }
         sentryApplicationSupportPath =
             [applicationSupportDirectory stringByAppendingPathComponent:@"io.sentry"];
+
+        // exclude our app support directory from backups
+        NSURL *url = [NSURL URLWithString:sentryApplicationSupportPath];
+        NSError *error;
+        if (![url setResourceValue:@YES forKey:NSURLIsExcludedFromBackupKey error:&error]) {
```

Comment:
This brings up the question of what to do if this line fails... maybe we should set `sentryApplicationSupportPath` to `nil` so that anything that wants to write there in the SDK after this cannot do so.

Reply:
I really doubt this will ever fail. 
If it fail its not our fault, we did what was asked of us.

Reply:
What about completely ditching `sentryApplicationSupportPath`? We only use it for `sentry_initializeAsyncLogFile`, which could also write to the cache I guess and for `enableAppLaunchProfiling`, which is still experimental. We only need to ensure that we clean up existing files in`sentryApplicationSupportPath`. As `enableAppLaunchProfiling` is still experimental we don't need to migrate it, IMO, and it's OK to delete the file for `sentry_initializeAsyncLogFile`, I guess @brustolin?

Then, we don't need to deal with this edge case. Also, the docs for [`NSURLIsExcludedFromBackupKey`](https://developer.apple.com/documentation/foundation/nsurlisexcludedfrombackupkey) sate:

> Use this property to exclude cache and other app support files that aren’t necessary in a backup. Set this property each time you save a file because some common file operations cause this property to reset to false.

It will be complicated to ensure to set this property for all files saved in `sentryApplicationSupportPath`. Even if we ensure to do that now, we can easily break this in the future. So I think it's better to ditch writing files to `sentryApplicationSupportPath`.

Reply:
> I really doubt this will ever fail.

Same, but if an API gives a way to report errors, you gotta handle the errors. Murphy's law.

> If it fail its not our fault, we did what was asked of us.

I don't see it this way. We're ultiimately responsible for our usage of any API we choose to use, even if we don't cover an edge case correctly.

Ugh, good catch with that doc @philipphofmann. I'm fine with changing this to NSCachesDirectory for now. 

As for how it can affect profiling, it's possible we lose the ability to profile anything other than a warm or prewarmed launch, because presumably a cold launch will only happen after something like an app update or device backup restore where caches may not already be available. I don't have a sense for how frequently NSCachesDirectory is wiped otherwise. We could maybe build telemetry for that, might even make a nice SDK feature. But I guess this is probably not used too much yet so probably wouldn't impact customers much, and we can always respond if they do ask for us to revert the behavior. I think ultimately we can get the NSApplicationSupportDirectory usage right if we wrap it correctly with SentryFileManager, should the need arise.

Reply:
> As for how it can affect profiling, it's possible we lose the ability to profile anything other than a warm or prewarmed launch, because presumably a cold launch will only happen after something like an app update or device backup restore where caches may not already be available.

I mean we can also keep it if required, but we need to ensure it's hard to shoot ourselves in the foot by correctly wrapping the usage. We shouldn't return the path but instead offer methods to read and write to `NSApplicationSupportDirectory` or something similar.